notebook/static/notebook/js/notebook.js

```diff
@@ -2844,7 +2844,100 @@ define([
             this._checkpoint_after_save = false;
         }
     };
-    
+
+    Notebook.prototype.save_notebook_as = function() {
+        // If current notebook has some changes, save them, or the copied notebook won't have them.
+        if (this.writable && this.dirty) {
+            this.save_notebook();
+        }
+        var that = this;
+        var current_dir = $('body').attr('data-notebook-path').split('/').slice(0, -1).join("/");
+        current_dir = current_dir? current_dir + "/": "";
+        var dialog_body = $('<div/>').append(
+            $('<p/>').addClass('save-message')
+                .text(i18n.msg._('Enter a notebook path relative to notebook dir'))
+        ).append(
+            $('<br/>')
+        ).append($('<label />').attr('for', 'save-as-dialog')
+            .html('<i class="fa fa-home"></i>')
+        ).append(
+            $('<input/>').attr('type','text').attr('size','25')
```

Comment:
This input area doesn't show up well for me. I think you're missing the `form-control` class that the rename dialog uses:

https://github.com/jupyter/notebook/blob/2aca6f98b0e46c9a3d635faf0c03abf113ff4089/notebook/static/notebook/js/savewidget.js#L83

Also, I hope we can get rid of the inline styling a couple of lines below.